mbed_targets/mbed_targets.py

```diff
@@ -58,11 +58,25 @@ def __init__(self) -> None:
             TargetDatabaseError: Failed to get the target data.
         """
         self._target_data: List[dict] = target_database.get_target_data()
+        self._target_iterator: Iterator = iter(self._target_data)
```

Comment:
This `self._target_iterator` is odd, as it's overwritten in `__iter__`, yet reused in `__next__`. I find this a bit confusing to follow.